src/edges_cal/spectra.py

```diff
@@ -581,14 +582,16 @@ def from_edges3(
         if freq_bin_size > 1:
             if frequency_smoothing == "bin":
                 q = tools.bin_array(q, size=freq_bin_size)
```

Comment:
I think the problem is that the flagged values will still be averaged in here. Unfortunately, there's a bit of a circular-dependency problem here, because the correct tools to use, that would incorporate flags etc., are already written but they're in the `edges_analysis.averaging` subpackage. We could patch them into `edges-cal`, or just do a hack here, but actually probably the right way to deal with this is to re-think how these packages are "ordered". Instead of edges-analysis depending on edges-cal, we should probably think of them having the inverse dependency structure, or even just merging the codes completely.